### PR TITLE
feat: adding CLEAN_KEYS support

### PR DESCRIPTION
**Pull request by harshilgajera-crest**
Added support for CLEAN_KEYS in Transforms.conf.
Added respective e2e test case for the same

Impact on Addons:

Now if Addons have have field names in transforms report which have non-alphanumeric chars then those chars would be first converted to _ and then tests would be generated as by default in splunk CLEAN_KEYS is set to true, so splunk also does the same thing.

Moreover if any report stanza has explicitly set CLEAN_KEYS=false then for those reports field conversion would not happen and tests would be generated as it is.

### CODE REVIEW

File: pytest_splunk_addon/addon_parser/props_parser.py [hunk 1]
```diff
@@ -57,6 +57,14 @@ def props(self) -> Optional[Dict]:
         self._props = self._conf_parser.item_dict()
         return self._props if self._props else None
 
+    def update_field_names(self, field_list) -> List:
+        """
+        update field names to remove all the non-alphanumeric chars and replace them with _
+        """
+        for index in range(len(field_list)):
```
Comment (artemrys) [inline pytest_splunk_addon/addon_parser/props_parser.py:64]: Can we do `for field in field_list`?

**Review by artemrys — COMMENTED** [review]
Can you please document the impact of this change on the exiting TAs?

File: pytest_splunk_addon/addon_parser/props_parser.py (hunk 1 above)
Comment (harshilgajera-crest) [reply pytest_splunk_addon/addon_parser/props_parser.py]: Done

File: pytest_splunk_addon/addon_parser/props_parser.py [hunk 2]
```diff
@@ -57,6 +57,14 @@ def props(self) -> Optional[Dict]:
         self._props = self._conf_parser.item_dict()
         return self._props if self._props else None
 
+    def update_field_names(self, field_list) -> List:
```
[the remark this answers is not in the document]
Comment (harshilgajera-crest) [reply pytest_splunk_addon/addon_parser/props_parser.py]: Done

File: pytest_splunk_addon/fields_tests/test_templates.py [hunk 3]
```diff
@@ -23,6 +23,7 @@
 from ..addon_parser import Field
```
Comment (kdoroszko-splunk) [inline pytest_splunk_addon/fields_tests/test_templates.py:23]: Maybe we could keep the proper order of imports? (move `from ..addon_parser import Field` to line 27, and remove blank line 29)?

Comment (harshilgajera-crest) [reply pytest_splunk_addon/fields_tests/test_templates.py]: okay, done

**Review by kdoroszko-splunk — APPROVED** [review]

File: tests/e2e/addons/TA_fiction/default/transforms.conf [hunk 4]
```diff
@@ -22,6 +22,23 @@ SOURCE_KEY = event_id
 DELIMS = "="
 FIELDS = server_contact_mode, dest
 
+# Component tested: REPORT, DELIM-FIELDS-SOURCE_KEY
+# Scenario:# Similar to the above scenario
+## ## server-contact-mode should be searched as server_contact_mode as CLEAN_KEYS = true by default
```
Comment (mkolasinski-splunk) [inline tests/e2e/addons/TA_fiction/default/transforms.conf:27]: CLEAN_KEYS is set to false here, so how should this field be looked for? Just description required small adjustment.

**Review by mkolasinski-splunk — CHANGES_REQUESTED** [review]
Minor changes needed in TC descritption

File: tests/e2e/addons/TA_fiction/default/transforms.conf (hunk 4 above)
Comment (harshilgajera-crest) [reply tests/e2e/addons/TA_fiction/default/transforms.conf]: done

**Review by mkolasinski-splunk — APPROVED** [review]

**Pull request merged**